backend/app/dataspace_configuration.py

```diff
@@ -0,0 +1,26 @@
+import httpx
+from app.settings import conf
+from fastapi import HTTPException
+
+DATASPACE_CONFIGURATION = None
+
+
+async def get_dataspace_configuration():
+    """
+    Fetch the dataspace configuration from the dataspace base domain
+    """
+    global DATASPACE_CONFIGURATION
+    if DATASPACE_CONFIGURATION:
+        return DATASPACE_CONFIGURATION
+
+    async with httpx.AsyncClient() as client:
+        try:
+            resp = await client.get(
+                f"https://{conf.DATASPACE_BASE_DOMAIN}/.well-known/dataspace/dataspace-configuration.json"
+            )
+        except httpx.HTTPError as exc:
+            raise HTTPException(
+                400, f"Error fetching dataspace configuration from {exc.request.url}"
```

Comment:
Fetching the dataspace configuration is not really a client side error (4xx) error, rather a server side error (5xx). So I'd suggest rather using 500 than 400 as the error code.